### DISCUSSION TITLE
Enhance the data viewer for Pandas dataframes in VSCode

### DISCUSSION THREAD

**Discussion by xareelee** [microsoft/vscode-jupyter#9867] [New Ideas]
When you run Jupyter/IPython in VSCode, you can view the Pandas dataframes variable in the data viewer:

![Screen Shot 2022-04-29 at 05 31 15](https://user-images.githubusercontent.com/804308/165849400-838eeb16-c7e2-4812-87cd-be040ee34611.png)

Feature requests for data viewer:

## 1. add info as `info()` in the data viewer

- `dtype` for each column
- non-null count
- unique count
- total memory usage

as `df.info()`:

![Screen Shot 2022-04-29 at 05 36 31](https://user-images.githubusercontent.com/804308/165849799-58e44417-7df8-421c-8dc4-4c7dcccfa061.png)

## 2. add a button to view `describe()` for columns

- you can select columns to show stat description.
- when pressing the button, display a popup view to show the results of `describe()` for selected columns (or all columns if there is no selection) 

as `df.describe()`:

![Screen Shot 2022-04-29 at 05 36 41](https://user-images.githubusercontent.com/804308/165849914-37564623-741c-4794-a670-569b2195f7ca.png)

## 3. provide a textfield to input the expression to apply onto the dataframe

You can add fluent dot syntax onto the dataframe and view the result on the fly. It's useful for data tidying and cleaning. 

The textfield looks like:

```python
# The below line is the input in the textfield; `(df)` is a prompt for the variable name
(df) {this is a placeholder for hints}
```

For example, you can type `.max()` in the textfield, it will update the data viewer for the output: 

```python
(df) .max()
```

![Screen Shot 2022-04-29 at 05 50 13](https://user-images.githubusercontent.com/804308/165853023-512a98bc-2a9f-4a3b-ac03-ab122e4c725a.png)

You can chain the methods the with fluent dot syntax:

```python
(df) .astype({'Open': 'int32', 'High': 'int32', 'Low': 'int32', 'Close': 'int32'}).drop(columns=['Volume'])
```

![Screen Shot 2022-04-29 at 05 56 51](https://user-images.githubusercontent.com/804308/165854540-72c3fc6f-57ae-45c8-a14e-0f4aa7d7b042.png)

Those would be very useful to view the Pandas dataframes in VSCode.

**Comment by rchiodo** [comment]
Thanks for the suggestion, I'll transfer it to an issue. 

cc @jmew
